tensorflow/contrib/slim/python/slim/learning.py

```diff
@@ -788,12 +788,15 @@ def train(train_op,
           sv.start_queue_runners(sess, [chief_queue_runner])
           sess.run(init_tokens_op)
         try:
-          while not sv.should_stop():
-            total_loss, should_stop = train_step_fn(sess, train_op, global_step,
-                                                    train_step_kwargs)
-            if should_stop:
-              logging.info('Stopping Training.')
-              break
+          try:
+            while not sv.should_stop():
+              total_loss, should_stop = train_step_fn(sess, train_op, global_step,
+                                                      train_step_kwargs)
+              if should_stop:
+                logging.info('Stopping Training.')
+                break
+          except errors.OutOfRangeError:
+            logging.info('Caught OutOfRangeError. Stopping Training.')
```

Comment:
It's probably a logging.error, although I'm not sure why you would want to catch that error and recover from it.